sycl/include/sycl/detail/spirv.hpp

```diff
@@ -1022,8 +1022,10 @@ ControlBarrier(Group, memory_scope FenceScope, memory_order Order) {
 template <typename Group>
 typename std::enable_if_t<
     ext::oneapi::experimental::is_user_constructed_group_v<Group>>
-ControlBarrier(Group, memory_scope FenceScope, memory_order Order) {
-#if defined(__SPIR__)
+ControlBarrier(Group g, memory_scope FenceScope, memory_order Order) {
+#if defined(__NVPTX__)
+  __nvvm_bar_warp_sync(detail::ExtractMask(detail::GetMask(g))[0]);
+#else
```

Comment:
I have no preference.  But what you've done here is consistent with other parts of DPC++, at least.  For example, the [sub-group implementation](https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/oneapi/sub_group.hpp#L306-L316) assumes that SPIR-V intrinsics will be supported.  I think this makes sense, because some SPIR-V intrinsics are implemented in libclc.